Based on patch from blackfin, except I've also removed the target
name from the directories.

https://sourceforge.net/tracker/?func=detail&aid=2888821&group_id=146814&atid=766244

--- STLport-5.2.1/build/Makefiles/gmake/targetdirs.mak
+++ STLport-5.2.1/build/Makefiles/gmake/targetdirs.mak
@@ -36,18 +36,18 @@
 OUTPUT_DIR_A_STLDBG    := $(OUTPUT_DIR_STLDBG)
 endif
 
-BASE_INSTALL_DIR       ?= /usr/local
+BASE_INSTALL_DIR       ?= $(DESTDIR)/usr/local
 
-BASE_INSTALL_LIB_DIR   ?= $(DESTDIR)${BASE_INSTALL_DIR}
-BASE_INSTALL_BIN_DIR   ?= $(DESTDIR)${BASE_INSTALL_DIR}
-BASE_INSTALL_HDR_DIR   ?= $(DESTDIR)${BASE_INSTALL_DIR}
+BASE_INSTALL_LIB_DIR   ?= ${BASE_INSTALL_DIR}
+BASE_INSTALL_BIN_DIR   ?= ${BASE_INSTALL_DIR}
+BASE_INSTALL_HDR_DIR   ?= ${BASE_INSTALL_DIR}
 
-INSTALL_LIB_DIR        ?= ${BASE_INSTALL_LIB_DIR}/${TARGET_NAME}lib
-INSTALL_LIB_DIR_DBG    ?= ${BASE_INSTALL_LIB_DIR}/${TARGET_NAME}lib
+INSTALL_LIB_DIR        ?= ${BASE_INSTALL_LIB_DIR}/lib
+INSTALL_LIB_DIR_DBG    ?= ${INSTALL_LIB_DIR}
 ifndef WITHOUT_STLPORT
-INSTALL_LIB_DIR_STLDBG ?= ${BASE_INSTALL_LIB_DIR}/${TARGET_NAME}lib
+INSTALL_LIB_DIR_STLDBG ?= ${INSTALL_LIB_DIR}
 endif
-INSTALL_BIN_DIR        ?= ${BASE_INSTALL_BIN_DIR}/${TARGET_NAME}bin
+INSTALL_BIN_DIR        ?= ${INSTALL_LIB_DIR}
 INSTALL_BIN_DIR_DBG    ?= ${INSTALL_BIN_DIR}_g
 ifndef WITHOUT_STLPORT
 INSTALL_BIN_DIR_STLDBG ?= ${INSTALL_BIN_DIR}_stlg
